src/release/eol.rs

```diff
@@ -0,0 +1,51 @@
+use anyhow::Context;
+use chrono::{Date, Datelike, NaiveDate, Utc};
+use std::convert::TryFrom;
+use ubuntu_version::{Codename, Version};
+
+pub enum EolStatus {
+    Ok,
+    Imminent,
+    Exceeded,
+}
+
+pub struct EolDate {
+    pub version: Version,
+    pub ymd:     (u32, u32, u32),
+}
+
+impl EolDate {
+    pub fn fetch() -> anyhow::Result<Self> {
+        let version = Version::detect().context("failed to detect current Ubuntu release")?;
+
+        let codename = match Codename::try_from(version) {
+            Ok(codename) => codename,
+            Err(()) => return Err(anyhow!("Invalid Ubuntu version: {}", version)),
+        };
+
+        Ok(Self { version, ymd: codename.eol_date() })
+    }
+
+    pub fn status(&self) -> EolStatus {
+        let (eol_year, eol_month, eol_day) = self.ymd;
+        let current = Utc::now().date();
+
+        if exceeded(current, eol_year as i32, eol_month, eol_day) {
+            EolStatus::Exceeded
+        } else if impending(current, eol_year as i32, eol_month) {
+            EolStatus::Imminent
+        } else {
+            EolStatus::Ok
+        }
+    }
+}
+
+#[inline]
+fn impending(current: Date<Utc>, eol_year: i32, eol_month: u32) -> bool {
+    current.year() == eol_year && current.month() == eol_month
```

Comment:
I think this check isn't going the full month in advance since it's checking only within `eol_month`. For example, 19.10 EOL's on July 17, 2020. When I set the system date to June 30, 2020, I do not receive the EOL message, but going to July 1st makes it show up.

Reply:
I'll convert the EOL date to a `Date<Utc>`, and then check if the duration between the EOL and current date is more than 30 days.

Reply:
This should be fixed now